math.go

```diff
@@ -127,25 +142,111 @@ func (l *Line) PointDistanceSquared(point Point) float32 {
 		(y0-(y1+t*(y2-y1)))*(y0-(y1+t*(y2-y1)))
 }
 
-// Returns the point where the two lines intersect
-func (l *Line) LineIntersection(l2 Line) Point {
-	x1 := l.P1.X
-	x2 := l.P2.X
-	x3 := l2.P1.X
-	x4 := l2.P2.X
+// Left Hand Normal
```

Comment:
Can we please use godoc style comments: https://blog.golang.org/godoc-documenting-go-code